software/controller/lib/hal/spi_chain.tpp

```diff
@@ -222,14 +207,42 @@ void DaisyChain<MaxSlaves>::TransmitNextCommand() {
   }
 
   if(data_to_send){
-    SetupReception(receive_buffer_, num_slaves_);
+    spi_->SetupReception(receive_buffer_, num_slaves_);
+    // In case cs has not been high for long enough, delay sending the command
+    Time now = SystemTimer::singleton().now();
+    if(now - last_cs_rise_ < min_cs_high_time_) {
+      SystemTimer::singleton().delay(now - last_cs_rise_);
+    }
     // Send command with CS low
-    SendCommand(send_buffer_, num_slaves_, true);
-    // Force CS to stay low for 350ns, in our case, the minimum wait is 1µs.
-    SystemTimer::singleton().delay(microseconds(1));
+    spi_->SendCommand(send_buffer_, num_slaves_, true);
   } else {
     command_buffer_count_ = 0;
     // wait for the next command, which will trigger a new transmission
     transmitting_data_ = false;
   }
 }
+template<size_t MaxSlaves, size_t MaxRequestsPerSlave>
```

Comment:
These last 2 functions would probably be really nice to test in isolation, if they were not private. That's why I'm still itching to take those arrays of queues and along with all functions that only do scanning or shuffling thereof (but nothing with spi back-end), and somehow encapsulate them.